tools/mo/openvino/tools/mo/graph/graph.py

```diff
@@ -1071,6 +1091,26 @@ def add_opoutput(graph: Graph, node_name: str, port: int, cut: bool = True, keep
                                                                    'keep_output_port': keep_output_port})
         opoutput_node.in_edge()['data_attrs'] = ['fw_tensor_debug_info']
 
+    if user_defined_name is not None and (graph.stage == 'front' or graph.stage is None):
+        # Following code adds user_defined_name to tensor names list
+        # Not applicable for middle stage
+        prev_op_tensor_names = set()
+        in_edge_attrs = opoutput_node.in_edge()
+        if 'fw_tensor_debug_info' in in_edge_attrs:
+            for _, tensor_name in opoutput_node.in_edge()['fw_tensor_debug_info']:
+                prev_op_tensor_names.add(tensor_name)
+        if user_defined_name not in prev_op_tensor_names:
+            # TODO: This can be optimized. Tensor names can be stored as set, which is initialized after model loading.
+            graph_tensor_names = graph.get_tensor_names_set()
+            if user_defined_name in graph_tensor_names:
+                log.warning('Could not add user defined output name {} to tensor names list of {} node as '
+                            'graph contains tensor name with same name.'.format(user_defined_name,
+                                                                                opoutput_node.soft_get('name')))
```

Comment:
nit: same comment is to replace with `has_tensor_name`

Reply:
Done.